### PR TITLE
[crfsuite] Add new port

### PR DESCRIPTION
**Pull request by ava-YangL**
http://www.chokkan.org/software/crfsuite/

### CODE REVIEW

**Review by Rastaban — APPROVED** [review]

File: ports/crfsuite/fix_msc_ver.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+index 6350eb1..e7f10f3 100644
+--- a/include/os.h
++++ b/include/os.h
+@@ -47,7 +47,9 @@
+ #define strdup      _strdup
+ #define open        _open
+ #define isfinite    _finite
+-#define snprintf    _snprintf
++#if defined(_MSC_VER) && _MSC_VER<1900
++#  define snprintf _snprintf
```
Comment (dan-shaw) [inline ports/crfsuite/fix_msc_ver.patch]: Ideally we want to minimize the patch file changes in the source. Is it be possible to disable this warning, perhaps by passing a compile definition in the `CMakeLists.txt`?

File: ports/crfsuite/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+)
+
+find_library(LBFGS_LIBRARY lbfgs)
+
+include_directories(lib/cqdb/include/)
+include_directories(include/)
+include_directories(win32/liblbfgs/)
+
+
+add_library (cqdb STATIC ${SOURCE_FILE_CQDB} ${HEADERS_FILE_CQDB})
```
Comment (dan-shaw) [inline ports/crfsuite/CMakeLists.txt]: Is this a static library? If so, we should add `vcpkg_check_linkage(ONLY_STATIC_LIBRARY)` in the `portfile`.

File: ports/crfsuite/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO chokkan/crfsuite
+    REF 0.12
+    SHA512   7cb02c1ac72613bafd639e1ef9a7ce7ddb0d36b2e7038c45f9745ad177344c3f20c7df48dfc1bdce364af1d0a85957a918ed8fee078232d5a2c91ab3796ee7f0
+    HEAD_REF master
+    PATCHES  fix_msc_ver.patch
+)
+
+message(STATUS "source path is : ${SOURCE_PATH}")
```
Comment (dan-shaw) [inline ports/crfsuite/portfile.cmake]: Remove status message

**Review by dan-shaw — COMMENTED** [review]
LGTM, some comments

File: ports/crfsuite/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	"swig/*.cpp"
+        "frontend/*.c"
+)
+file(GLOB HEADERS_FILE
+	"include/*.h"        
+        "include/*.hpp"
+        "frontend/*.h"  
+)
+
+find_library(LBFGS_LIBRARY lbfgs)
```
Comment (dan-shaw) [inline ports/crfsuite/CMakeLists.txt]: Should be `find_package`.

Comment (ava-YangL) [reply ports/crfsuite/CMakeLists.txt]: Could you please tell me why we need "find_package" here?   lbfgs could be found in vcpkg,  If I change “find_library” to "find_package", I will get a cmake error.  Thanks~

File: ports/crfsuite/fix_msc_ver.patch (hunk 1 above)
Comment (ava-YangL) [reply ports/crfsuite/fix_msc_ver.patch]: It seems that I could't find a good solution for this question

Comment (ava-YangL) [reply ports/crfsuite/fix_msc_ver.patch]: Hi, this problem is solved, Thanks.

File: ports/crfsuite/CMakeLists.txt (hunk 4 above)
Comment (dan-shaw) [reply ports/crfsuite/CMakeLists.txt]: On second thought, it should be okay. I see a couple of other ports using `find_library`.

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**